src/bus/index.js

```diff
@@ -12,13 +15,73 @@ class Bus {
    */
   constructor(sdk, request) {
     const baseUrl = `${sdk.config.audiences.bus.host}`;
-
+    const baseSocketUrl = `${sdk.config.audiences.bus.socket}`;
+    this._baseSocketUrl = baseSocketUrl;
     this._baseUrl = baseUrl;
     this._request = request;
     this._sdk = sdk;
+    this._sockets = {};
 
     this.channels = new Channels(sdk, request, baseUrl);
   }
+
+  /**
+   * Connects to the message bus via websocket.
+   * If a connection already exists for that organization id, the connection is returned, otherwise a new connection is created.
+   *
+   * @param {string} organizationId UUID corresponding with an organization
+   *
+   * @returns {Promise}
+   * @fulfill {Socket}
+   * @reject {object} The error event
+   *
+   * @example
+   * contxtSdk.bus.connect('4f0e51c6-728b-4892-9863-6d002e61204d')
+   * .then((socket) => {
+   *    console.log(socket);
+   * })
+   * .catch((event) => {
+   *    console.log(event);
+   * })
+   */
+  connect(organizationId) {
+    return new Promise((resolve, reject) => {
```

Comment:
This should probably be inverted -- do the `new Promise` in the `then` so you don't have to handle the `catch` for if getting the current api token fails